src/AssocList.mo

```diff
@@ -1,47 +1,54 @@
 /// Map implemented as a linked-list of key-value pairs ("Associations").
-///
-/// NOTE: This map implementation is mainly used as underlying buckets for other map
-/// structures. Thus, other map implementations are easier to use in most cases.
+/// 
+/// :::note [Usage context]
+/// 
+/// This map implementation primarily serves as the underlying bucket structure for other map types. In most cases, those higher-level map implementations are easier to use.
+/// :::
+/// 
+/// :::note [Assumptions]
+/// 
+/// Runtime and space complexity assumes that `combine`, `equal`, and other functions execute in `O(1)` time and space.
+/// :::
 
 import List "List";
 
 module {
-  /// Import from the base library to use this module.
-  ///
-  /// ```motoko name=import
-  /// import AssocList "mo:base/AssocList";
-  /// import List "mo:base/List";
-  /// import Nat "mo:base/Nat";
-  ///
-  /// type AssocList<K, V> = AssocList.AssocList<K, V>;
-  /// ```
-  ///
-  /// Initialize an empty map using an empty list.
-  /// ```motoko name=initialize include=import
-  /// var map : AssocList<Nat, Nat> = List.nil(); // Empty list as an empty map
-  /// map := null; // Alternative: null as empty list.
-  /// map
-  /// ```
+  ///  Import from the base library to use this module.
```

Comment:
Indentation seems to have an extra space for many of the doc comments. Was this intentional? Otherwise, it might be possible to fix these using a multi-line regex find/replace.